Geometry/TrackerGeometryBuilder/interface/phase1PixelTopology.h

```diff
@@ -27,17 +28,20 @@ namespace phase1PixelTopology {
 
   constexpr uint32_t numberOfModules = 1856;
   constexpr uint32_t numberOfLayers = 10;
-  constexpr uint32_t layerStart[numberOfLayers + 1] = {0,
-                                                       96,
-                                                       320,
-                                                       672,  // barrel
-                                                       1184,
-                                                       1296,
-                                                       1408,  // positive endcap
-                                                       1520,
-                                                       1632,
-                                                       1744,  // negative endcap
-                                                       numberOfModules};
+#ifdef __CUDA_ARCH__
+  __device__
+#endif
```

Comment:
Looks like we should "abstract" this approach and make it easier to declare `constexpr` constants that work both on the host and device ?

Of course this PR is ok as it is, I'm just thinking how we can improve things later.

Maybe something like
```c++
#ifdef __CUDA_ARCH__
#define HOST_DEVICE_AGGREGATE_CONSTANT __device__ constexpr
#else
#define HOST_DEVICE_AGGREGATE_CONSTANT constexpr
#endif
```
?

Or maybe a simpler name :-)

Reply:
I did not want to introduce dependencies in this file in this moment (is used also in code that does not depend on Heterogeous)

Reply:
Opened https://github.com/cms-sw/cmssw/issues/36311 to remember about this.